src/neurotechdevkit/imaging/beamform.py

```diff
@@ -0,0 +1,356 @@
+    tx_delays: npt.NDArray[np.float_],
+    freq_sampling: float,  # Hz
+    freq_carrier: float,  # Hz
+    start_time: float = 0,  # s
+    speed_sound: float = 1540,  # m/s
+    f_number: Optional[float] = 1.5,
+    width: Optional[float] = None,  # m
+    bandwidth: Optional[float] = 0.75,
+    method: InterpolationMethod = InterpolationMethod.LINEAR,
+) -> csr_array:
+    """
+    Calculate the delay-and-sum sparse matrix for beamforming I/Q signals.
+
+    Returns:
+        delay-and-sum matrix. Shape: (num_pixels, num_samples*num_channels)
+
+    Assumptions:
+        - The input signal is in-phase/quadrature (I/Q) signals.
+        - The input signal is sampled at a constant rate.
+        - The sensor array is linear.
+        - The sensor array is parallel to the x-axis and centered at (0, 0).
+
+    Notes:
+        - Alternatively, delay-and-sum can be implemented using an explicit
+            for-loop over the delays. However, Python for-loops are much
+            slower than  optimized sparse matrix multiplication.
+        - Memory usage for the delay-and-sum matrix is:
+            O(num_pixels * num_channels * num_interpolation_points).
+
+    For details on delay-and-sum by matrix-multiplication:
+        https://doi.org/10.1016/j.ultras.2020.106309
+    """
+    # Check inputs
+    assert num_time_samples > 0, "Number of time samples must be positive."
+    assert num_channels > 0, "Number of channels/elements must be positive."
+    assert (x.size > 0) and (z.size > 0), "Number of pixels must be positive."
+    assert x.ndim == 2, "Expected x to have shape (width_pixels, depth_pixels)."
+    assert z.ndim == 2, "Expected z to have shape (width_pixels, depth_pixels)."
+    assert x.shape == z.shape, "Expected image grid x and z to have the same shape."
+    assert len(tx_delays) == num_channels, "Expected one transmit-delay per channel."
+    if f_number is None:
+        # Optimize f-number based on other parameters
+        assert width is not None, "Element width is required to estimate f-number."
+        assert (
+            bandwidth is not None
+        ), "Element bandwidth is required to estimate f-number."
+        assert 0 < bandwidth < 2, "Fractional bandwidth at -6dB must be in (0, 2)."
+
+        # Eq. 14 of https://doi.org/10.1016/j.ultras.2020.106309
+        wavelength_min = speed_sound / (freq_carrier * (1 + bandwidth / 2))
+
+        # Eq. 11 of https://doi.org/10.1016/j.ultras.2020.106309
+        def directivity(theta, width, wavelength):
+            return np.cos(theta) * np.sinc(width / wavelength * np.sin(theta))
+
+        optimal_directivity = 0.71
+
+        # Eq. 13 of https://doi.org/10.1016/j.ultras.2020.106309
+        result = minimize_scalar(
+            lambda theta: np.abs(
+                directivity(theta, width, wavelength_min) - optimal_directivity
+            ),
+            bounds=(0, np.pi / 2),
+        )
+        alpha = result.x
+        f_number = 1 / (2 * np.tan(alpha))
```

Comment:
does it make sense to move this to another function?

Reply:
Yes, updated! See `_calculate_receptive_fields`